production/dpl-workflow.sh

```diff
@@ -259,74 +275,63 @@ if [ $CTFINPUT == 0 ]; then
+  has_detector_reco MFT && WORKFLOW+="o2-mft-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --clusters-from-upstream $DISABLE_MC --disable-root-output --pipeline mft-tracker:$N_MFTTRK --mft-dictionary-path $FILEWORKDIR | "
+  has_detector_reco FDD && WORKFLOW+="o2-fdd-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC | "
+  has_detector_reco FDD && WORKFLOW+="o2-fv0-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC | "
+  has_detector_reco ZDC && WORKFLOW+="o2-zdc-digits-reco $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC | "
 fi
 
-# ---------------------------------------------------------------------------------------------------------------------
-# Workflows disabled in async mode
-if [ $CTFINPUT == 0 ]; then
-  has_detector PHS && WORKFLOW+="o2-phos-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --input-type raw --output-type cells --disable-root-input --disable-root-output --pipeline PHOSRawToCellConverterSpec:$N_F_REST $DISABLE_MC | "
-  has_detector CPV && WORKFLOW+="o2-cpv-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --input-type raw --output-type clusters --disable-root-input --disable-root-output --pipeline CPVRawToDigitConverterSpec:$N_F_REST,CPVClusterizerSpec:$N_F_REST $DISABLE_MC | "
-  has_detector EMC && WORKFLOW+="o2-emcal-reco-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --input-type raw --output-type cells --disable-root-output $DISABLE_MC --pipeline EMCALRawToCellConverterSpec:$N_EMC | "
-  if ! workflow_has_parameter CTF_ONLY; then
-    # Reconstruction workflows
-    if [ $SYNCMODE == 1 ]; then # Otherwise already present in async setup
-      has_detectors TOF TRD TPC ITS && WORKFLOW+="o2-tof-matcher-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC --track-sources \"ITS-TPC\" --pipeline tof-matcher:$N_TOFMATCH | "
-    fi
-  fi
+has_detectors_reco MFT MCH && has_detector_matching MFTMCH && WORKFLOW+="o2-globalfwd-matcher-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output $DISABLE_MC --pipeline globalfwd-track-matcher:$N_F_REST | "
+has_detectors_reco ITS TPC TRD TOF FT0 MCH MFT && has_detector_matching PRIMVTX && WORKFLOW+="o2-primary-vertexing-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" $DISABLE_MC --disable-root-input --disable-root-output --validate-with-ft0 --pipeline primary-vertexing:$N_F_REST | "
+has_detectors_reco ITS TPC TRD TOF FT0 MCH MFT && has_detector_matching SECVTX && WORKFLOW+="o2-secondary-vertexing-workflow $ARGS_ALL --configKeyValues \"$ARGS_ALL_CONFIG\" --disable-root-input --disable-root-output --pipeline secondary-vertexing:$N_F_REST | "
```

Comment:
I think we also need to compose incrementally the input sources for P/S vertexing (in principle, can be the same, collecting all available track).
I would not forbid the vertexing if all {ITS TPC TRD TOF FT0 MCH MFT} are not there. Technically, ITS alone is enough  (and necessary) to run primary vertexing,  for the secondary one it is ITS+TPC (and soon MFT will join for forward secondaries).

Reply:
Sure, but since the vertexers were for async only I didn't spend much time so far. Do the vertexers have a `--track-sources` option or similar?

Reply:
`o2-primary-vertexing-workflow` has 2 input sources lists:
`--vertexing-sources <used_for_PVertex_finding>`
`--vetex-track-matching-sources <sources to be associated with vertices matching time-wise>`

`o2-secondary-vertexing-workflow` has simply 
`--vertexing-sources <sources used for v0 and cascade finding>`